File: sqlparser/src/sql/SqlGetDeletedStatement.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

import core.ParserCore;
import core.SqlExceptions;

/**
 * Syntax:
 *         GETDELETED FROM TABLENAME WHERE PARAM= ? [...]
 */
public class SqlGetDeletedStatement extends SqlStatement{
  private SqlConditionNode condition;

  public /*#override#*/ void accept(ISqlQueryVisitor visitor) throws Exception {

  }

  public /*#override#*/  SqlConditionNode getCriteria() {
    return condition;
  }

  public /*#override#*/  void setCriteria(SqlConditionNode criteria) {
    condition = criteria;
  }
  
  public /*#override #*/void addCondition(SqlConditionNode cond) {
    if (condition == null) {
      condition = cond;
    } else {
      condition = new SqlCondition(condition, SqlLogicalOperator.And, cond);
    }
  }

  public SqlGetDeletedStatement(SqlTokenizer tokenizer, Dialect dialectProcessor) throws Exception {
    super(dialectProcessor);
    tokenizer.NextToken();
    tokenizer.EnsureNextToken("FROM");
    parseTableName(tokenizer);
    parseQuery(tokenizer);
  }

  public SqlGetDeletedStatement(Dialect dialectProcessor) {
    super(dialectProcessor);
  }

  @Override
  public Object clone() {
    SqlGetDeletedStatement obj = new SqlGetDeletedStatement(null);
    obj.copy(this);
    return obj;
  }

  @Override
  protected void copy(SqlStatement obj) {
    super.copy(obj);
    SqlGetDeletedStatement o = (SqlGetDeletedStatement)obj;
    this.condition = o.condition == null ? null : (SqlConditionNode)o.condition.clone();
  }

  private void parseQuery(SqlTokenizer tokenizer) throws Exception {
    SqlToken next = tokenizer.LookaheadToken2();
    if (next.equals("WHERE")) {
      this.condition = ParserCore.ParseWhere(tokenizer, this);
    }
  }
  
  private void parseTableName(SqlTokenizer tokenizer) throws Exception {
    SqlToken nextToken = tokenizer.LookaheadToken2();
    if(nextToken.IsEmpty()){
      throw tokenizer.MalformedSql(SqlExceptions.localizedMessage(SqlExceptions.ENDED_BEFORE_TABLENAME));
    }
    if(nextToken.equals("WHERE")){
      throw tokenizer.MalformedSql(SqlExceptions.localizedMessage(SqlExceptions.EXPECTED_TABLENAME_BEFORE_WHERE));
    }

    String [] period = ParserCore.parsePeriodTableName(tokenizer, this.getDialectProcessor());
    this.tableName = period[2];
    this.table = new SqlTable(period[0], period[1], tableName, tableName);
  }
}

File: sqlparser/src/sql/csgen/SqlGetDeletedStatement.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


/**
 * Syntax:
 *         GETDELETED FROM TABLENAME WHERE PARAM= ? [...]
 */
public class SqlGetDeletedStatement : SqlStatement{
  private SqlConditionNode condition;

  public override void Accept(ISqlQueryVisitor visitor) {

  }

  public override  SqlConditionNode GetCriteria() {
    return condition;
  }

  public override  void SetCriteria(SqlConditionNode criteria) {
    condition = criteria;
  }
  
  public override void AddCondition(SqlConditionNode cond) {
    if (condition == null) {
      condition = cond;
    } else {
      condition = new SqlCondition(condition, SqlLogicalOperator.And, cond);
    }
  }

  public SqlGetDeletedStatement(SqlTokenizer tokenizer, Dialect dialectProcessor) : base(dialectProcessor) {
    tokenizer.NextToken();
    tokenizer.EnsureNextToken("FROM");
    ParseTableName(tokenizer);
    ParseQuery(tokenizer);
  }

  public SqlGetDeletedStatement(Dialect dialectProcessor) : base(dialectProcessor) {
  }

  public override Object Clone() {
    SqlGetDeletedStatement obj = new SqlGetDeletedStatement(null);
    obj.Copy(this);
    return obj;
  }

  protected override void Copy(SqlStatement obj) {
    base.Copy(obj);
    SqlGetDeletedStatement o = (SqlGetDeletedStatement)obj;
    this.condition = o.condition == null ? null : (SqlConditionNode)o.condition.Clone();
  }

  private void ParseQuery(SqlTokenizer tokenizer) {
    SqlToken next = tokenizer.LookaheadToken2();
    if (next.Equals("WHERE")) {
      this.condition = ParserCore.ParseWhere(tokenizer, this);
    }
  }
  
  private void ParseTableName(SqlTokenizer tokenizer) {
    SqlToken nextToken = tokenizer.LookaheadToken2();
    if(nextToken.IsEmpty()){
      throw tokenizer.MalformedSql(SqlExceptions.LocalizedMessage(SqlExceptions.ENDED_BEFORE_TABLENAME));
    }
    if(nextToken.Equals("WHERE")){
      throw tokenizer.MalformedSql(SqlExceptions.LocalizedMessage(SqlExceptions.EXPECTED_TABLENAME_BEFORE_WHERE));
    }

    string [] period = ParserCore.ParsePeriodTableName(tokenizer, this.GetDialectProcessor());
    this.tableName = period[2];
    this.table = new SqlTable(period[0], period[1], tableName, tableName);
  }
}
}

